app/helpers/lightbox_helper.rb

```diff
@@ -17,7 +17,7 @@ def lightbox_caption_html(lightbox_data)
     obs_data = lightbox_data[:obs_data]
     html = []
     if obs_data[:id].present?
-      html = lightbox_obs_caption(html, obs_data, lightbox_data[:identify])
+      lightbox_obs_caption(html, obs_data, lightbox_data[:identify])
```

Comment:
Not sure this is equivalent

Reply:
By accident it is.  The point is that lightbox_obs_caption just adds stuff to html, it doesn't really return anything.  Apparently `html << blah` returns html as a side-effect, so it's okay that you were replacing html with, well, html.  But it bit me in the foot and confused me mightily when I'd commented out everything in lightbox_obs_caption.  If you really want it to take the form of lightbox_obs_caption returning something, instead of changing an argument in place, then please rewrite it as:

```
html += lightbox_obs_caption(obs_data, blah)
...
def lightbox_obs_caption(obs_data, identify)
  html = []
  if blah
    html << blah
    html << blah
  end
  html << blah
  html << blah
  html
end
```

I think that would be much less confusing, personally.

Tiny point, but it just confused and wasted some of my time last night, and this tiny change would have avoided that.

Reply:
Thanks. That `=` was definitely a mistake

Reply:
Oh wait, no, it was just confusing style. Now i get what you're saying.

Reply:
Good!

(But I do love the `html << lightbox_obs_caption(html)` idea.  I wondered briefly if the universe would implode if you tried to push an Array inside of itself. ;)